### PR TITLE
Add android progress callback

### PR DESCRIPTION
**Pull request by sbycrosz**
Hi, 

I added a progress callback to the scan function for better scanning UX 🎉 

Pretty much copy-pasted from [this commit](https://github.com/EstebanFuentealba/react-native-passport-reader/commit/e495ccbcc728609f2e0440ad673acc0307da3c8b)

Cheers,
Sam

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "name": "react-native-passport-reader",
   "version": "1.0.3",
   "description": "read the NFC chip in a passport",
-  "main": "index.android.js",
```
Comment (mvayngrib) [inline package.json:5]: ~why this change?~ nm, i see the index.ios.js

File: README.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ If your app will not function without nfc capabilities, set `android:required` a
 ## Usage
 ```js
 import PassportReader from 'react-native-passport-reader'
-// { scan, cancel, isSupported }
+// { scan, cancel, isSupported, constants }
```
Comment (mvayngrib) [inline README.md]: while we're here, i think `isSupported` is part of `constants` rather than on its own
```suggestion
// { scan, cancel, constants }
```

File: index.android.js [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+  
+  try {
+    const result = await RNPassportReader.scan({ documentNumber, dateOfBirth, dateOfExpiry, quality })
+    
+    listener.remove()
+    return result
+  } catch (error) {
+    listener.remove()
+    throw error
+  }
```
Comment (mvayngrib) [inline index.android.js]: slightly simpler pattern:
```suggestion
  try {
    return await RNPassportReader.scan({ documentNumber, dateOfBirth, dateOfExpiry, quality })
  } finally {
    listener.remove()
  }
```

**Comment by mvayngrib** [comment]
thanks for the PR @sbycrosz ! I'm unable to test this so I'm going to rely on you to do it after we're done here, is that ok? 😛

**Comment by sbycrosz** [comment]
Okay, seems to be working fine on my project 👍 

But this library does need an example project hahaha
